controllers/app.js

```diff
@@ -3,78 +3,78 @@
-        const arr = [];
-        const book = $scope.newIndex.books;
-        book.forEach((bookTitle) => {
-            arr.push(bookTitle.title);
-        });
-        // console.log(arr);
-        return arr;
-    };
+  const indexCount = (n) => {
+    const arr = [];
+    for (let i = 0; i < n; i += 1) {
+      arr.push(i);
+    }
+    return arr;
+  };
+  const titlesList = () => { // eslint-no constant
+    const arr = [];
+    const book = $scope.newIndex.books;
+    book.forEach((bookTitle) => {
+      arr.push(bookTitle.title);
+    });
+    return arr;
+  };
 
-    $scope.uploader = () => {
-        const fileDoc = document.getElementById('myJsonFile').files[0];
-        $scope.fileDocName = fileDoc.name;
-        if (typeof(fileDoc !== 'Blob')) {
-            $scope.uploaderError = 'select and upload a json object and then create index'; // eslint-no constant condition
-        }
-        // $scope.newIndex = new InvertedIndexClass();
-        const reader = new FileReader();
-        reader.readAsText(fileDoc);
-        reader.onload = (event) => {
-            $scope.readybooks = event.target.result;
-            $scope.validationResult = $scope.newIndex.validFiles($scope.readybooks);
+  $scope.uploader = () => {
+    const fileDoc = document.getElementById('myJsonFile').files[0];
+    $scope.fileDocName = fileDoc.name;
+    if (typeof (fileDoc !== 'Blob')) {
+      $scope.uploaderError = 'select and upload a json object and then create index'; // eslint-no constant condition
+    }
+    const reader = new FileReader();
+    reader.readAsText(fileDoc);
+    reader.onload = (event) => {
+      $scope.readybooks = event.target.result;
+      $scope.validationResult = $scope.newIndex.validFiles($scope.readybooks);
 
-            $scope.$apply(() => {
-                if ($scope.validationResult[0]) {
-                    $scope.allValidatedBooks[$scope.fileDocName] = $scope.readybooks;
-                }
-            });
-            alert($scope.validationResult[1]); // eslint-disable-line no-alert
-        };
-    };
-    $scope.submitIndex = () => {
-        if ($scope.validationResult[0] === true) {
-            $scope.indexOfWords = $scope.newIndex.createIndex($scope.indexToBeCreated,
-                JSON.parse($scope.allValidatedBooks[$scope.indexToBeCreated]));
-            $scope.titles[$scope.indexToBeCreated] = titlesList($scope.newIndex.books.length);
-            $scope.arrayLength[$scope.indexToBeCreated] = indexCount($scope.newIndex.books.length);
-            $scope.showIndex = true;
-            $scope.showSearch = false;
+      $scope.$apply(() => {
+        if ($scope.validationResult.status === true) {
+          $scope.allValidatedBooks[$scope.fileDocName] = $scope.readybooks;
+          $scope.uploaderError = 'success! Please create an index of the uploaded book below';
+        } else {
+          alert($scope.validationResult.statusMessage); // eslint-disable-line no-alert
```

Comment:
fixed. I mentioned it yesterday that the errors were for just error messages. Taken them out now and they display just beneath the upload button.